### PR TITLE
ECO-15

### PR DESCRIPTION
**Pull request by alistairjudson**
Updated to have more consistent testability.

### CODE REVIEW

**Comment by geototti21** [comment]
Update readme file please

File: tokengen.go [hunk 1]
```diff
@@ -22,11 +20,22 @@ type TokenProvider interface {
 	GenerateToken() (string, error)
 }
 
+func New(charset string, length int) (Tokengen, error) {
```
Comment (Connor-McGuinness) [inline tokengen.go]: This'll need a comment, and it'd be worth enabling golint on build.

File: tokengen.go [hunk 2]
```diff
@@ -35,84 +47,16 @@ type Tokengen struct {
...
-	output = append(output, runes...)
-	return output, nil
-}
-
-func bytesPerRuneIndex(numRunes int) (bytesPerRune, maxValue int) {
-	maxValue = 1
-	// iterate until we find a multiple of 256 greater than the number of runes
-	for bytesPerRune = 0; maxValue <= numRunes; bytesPerRune++ {
-		maxValue *= (1 << 8)
+		return ``, err
```
Comment (jkerr123) [inline tokengen.go]: Why return the empty backticks instead of nil?

Comment (alistairjudson) [inline tokengen.go]: because you can't return nil as string...

Comment (jkerr123) [inline tokengen.go]: Makes sense 😛

**Pull request merged**